.github/workflows/release.yaml

```diff
@@ -44,44 +41,30 @@ jobs:
-#          docker push public.ecr.aws/opsterio/opensearch-operator:release
-#       - name: Create Release
-#         id: create_release
-#         uses: actions/create-release@v1
-#         env:
-#           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-#         with:
-#           tag_name: ${{ github.ref }}
-#           release_name: ${{ github.ref }}
-#           draft: false
-#           prerelease: false
-#       - name: Copy-crd-to-chart-dir
-#         run: |
-#           git checkout -b ${RELEASE_VERSION}-release
-#           cd opensearch-operator
-#           cp ./config/crd/bases/opensearch.opster.io_opensearchclusters.yaml ../charts/opensearch-operator/templates/opensearchclusters.opensearch.opster.io-crd.yaml
-#           sed -i -e 's/^appVersion.*/appVersion: '$RELEASE_VERSION'/' ../charts/opensearch-operator/Chart.yaml
-#           sed -i -e 's/tag.*/tag: '$RELEASE_VERSION'/' ../charts/opensearch-operator/values.yaml
-#           git config user.name "GitHub Actions Bot"
-#           git config user.email "<>"
-#           git add ../charts/opensearch-operator/templates/opensearchclusters.opensearch.opster.io-crd.yaml
-#           git add ../charts/opensearch-operator/Chart.yaml
-#           git add ../charts/opensearch-operator/values.yaml
-#           git commit -m "Update release appVersion on Chart ${RELEASE_VERSION}"
-#           git pull main:${RELEASE_VERSION}-release
-#           git push --set-upstream origin ${RELEASE_VERSION}-release
 
-#          git request-pull ${RELEASE_VERSION}-release https://github.com/Opster/opensearch-k8s-operator.git main
+      - name: Make sure helm chart has correct versions
+        run: |
+          sed -i -e 's/^appVersion:.*/appVersion: '$RELEASE_VERSION'/' charts/opensearch-operator/Chart.yaml
+          sed -i -e 's/^version:.*/version: '$RELEASE_VERSION'/' charts/opensearch-operator/Chart.yaml
+
+      - name: Install Helm
+        uses: azure/setup-helm@v1
+        with:
+          version: v3.8.1
 
-###### After build and push the images we can now prepper the chart release - just after the chart release will create a new helm-release workflow will run.
-###### that's a problem because we need to marge the version change commit first so the chart will be created with the new version.
-#      - name: Create-Helm-Release
-#        id: create_helm_release
-#        uses: actions/create-release@latest
-#        env:
-#          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # This token is provided by Actions, you do not need to create your own token
-#        with:
-#          tag_name: helm-$RELEASE_VERSION
-#          release_name: helm-$RELEASE_VERSION
-#          draft: false
-#          prerelease: false
+      - name: Publish helm chart
+        uses: stefanprodan/helm-gh-pages@master
+        with:
+          charts_dir: charts/
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Create Release
```

Comment:
There is no option to add the Changelog and release information as a parameter in the workflow?

Reply:
Most release actions I found only allow to read the text of a release from a file. So we would need to maintain a changelog in the code (which we will likely often forget). When creating/editing a release github provides the option to generate a changelog, so I thought this could remain a manual step (and imo a human should check and edit the release information and changelog anyway so it's not really extra work).